srv_hijacker/srv_hijacker.py

```diff
@@ -59,15 +60,66 @@ def patched_f(host, port, family=0, type=0, proto=0, flags=0):
     return patched_f
 
 
-def hijack(host_regex, srv_dns_host=None, srv_dns_port=None):
+class PatchError(Exception):
+    pass
+
+
+def _patch_psycopg2(host_regex, srv_resolver):
+    try:
+        import psycopg2
+        from psycopg2.extensions import parse_dsn, make_dsn
+    except ImportError as e:
+        raise PatchError(f"failed to import psycopg2. {e}")
+
+    def patch_psycopg2_connect(fn):
+        @wraps(fn)
+        def wrapper(dsn, connection_factory=None, *args, **kwargs):
+            config = parse_dsn(dsn)
+            host = config.get("host")
+            if host is None:
+                host = kwargs.get("host")
+
+            if host is None:
+                # Host may be left out to use localhost or
+                # possibly set using environment variables, nothing
+                # we can do in either case.
+                logger.error(
+                    "'host' parameter is not present in call to psycopg2.connect, "
+                    "DNS resolution might not work properly"
+                )
+
+                return fn(dsn, connection_factory, *args, **kwargs)
+
+            if re.search(host_regex, host):
+                logger.debug("Host %s matched SRV regex, resolving", host)
+                host, port = resolve_srv_record(host, srv_resolver)
+                config["host"] = host
+                config["port"] = port
+
+            dsn = make_dsn(**config)
+
+            return fn(dsn, connection_factory, *args, **kwargs)
+
+        return wrapper
+
+    psycopg2._connect = patch_psycopg2_connect(psycopg2._connect)
+
+
+PATCHABLE_LIBS = {
+    "psycopg2": _patch_psycopg2,
+}
+
+
+def hijack(host_regex, srv_dns_host=None, srv_dns_port=None, libraries_to_patch=None):
     """
     Usage:
 
     ```
     srv_hijacker.hijack(
         host_regex=r'service.consul$',
         srv_dns_host='127.0.0.1',
-        srv_dns_port=8600
+        srv_dns_port=8600,
+        libraries_to_patch=['psycopg2']
```

Comment:
Can we remove the default here? I don't think it's wise to assume that everyone has psycopg2 installed by default, probably better if this was opt-in..

Reply:
Just to be clear - I'm suggesting that the default be an empty list.

Reply:
This is the documentation comment. function signature is still empty by default. I updated the docblock to demonstrate the usage of `libraries_to_patch`

Reply:
Ah, sorry - didn't notice that